vm/inc/ubpf.h

```diff
@@ -26,6 +26,15 @@ typedef uint64_t (*ubpf_jit_fn)(void *mem, size_t mem_len);
 struct ubpf_vm *ubpf_create(void);
 void ubpf_destroy(struct ubpf_vm *vm);
 
+/*
+ * Enable / disable bounds_check
+ *
+ * Bounds check is enabled by default, but it may be too restrictive
+ * Pass true to enable, false to disable
+ * Returns previous state
+ */
+bool toggle_bounds_check(struct ubpf_vm *vm, bool enable);
```

Comment:
I'm not sure I understand the use case, but I'd expect this to be a compile-time option.

Reply:
With this toggle I can change behavior not affecting another project using the same library. Compile time option would not make it.
Also I made toggle in ubpf_vm so that several VMs may be created in the same space and behave differently.

Reply:
For the record, rbpf handles this kind of specific use cases by allowing users to register their own verifier in lieu of the one shipped by default. Not sure if this is suitable for uBPF, but in this case it would allow redefining bound checks locally.

Reply:
> A block of memory (let's say sk_buff) may be allocated somewhere and pointer passed into a bpf function (compiled separately and loaded by uBPF). Currently it causes "uBPF error: out of bounds"

Ok, I see :+1: 
Please note this function will stop having any effect if we introduce a complete verifier later on. We can rediscuss it if/when that happens.

> For the record, rbpf handles this kind of specific use cases by allowing users to register their own verifier in lieu of the one shipped by default. 

@qmonnet uBPF does bounds checks at runtime, so it isn't really part of the verifier.

Reply:
Thank you!